CONTEXT.md

```diff
@@ -0,0 +1,74 @@
+# Context command
+
+Context files are used to define deplployment enviroments. Multiple contexts representing different enviroments can be created with each context specifying an appropriate provider  and a set of properties required/understood by that provider with the ability to switch between them. Contexts are persisted in individual files under the `.fn` directory which, if not present, is created on launch.
+
+```
+~ 
+  .fn
+     config.yaml
+     contexts
+        default.yaml
+```
+
+The `config.yaml` contains the name of currently selected context, when first created the current context is not set:
+
+ ```
+ current-context: ""
+ ``` 
+
+Within the `default.yaml` default values are set: 
+```
+api_url: http://localhost:8080/v1
+provider: default
+registry: ""
+```
+
+* api_url - the fn-server endpoint.
+* provider - the a specific provider which identifies a set of properties required/understood by that provider.
+* registry - the Docker registry username to push images to 
+[registry.hub.docker.com/`registry`].
+
+### Listing Contexts
+The `fn context` command accepts either `l` or `list` to view a list of contexts.
+```
+$ fn context list
+```
+
+### Creating a Context 
+
+To create a context use `c` or `create`. The context file will be created with default values but option flags can be used override.
+
+```
+$ fn context create <context>
+$ fn context create <context> --api-url foo --provider bar --registry <dockerhub-username>
+```
+
+### Setting a Context 
+
+To set a context use `s` or `set`.
+
+```
+$ fn context set <context>
```

Comment:
Would `use` be a better name than `set`? Set feels like it should be setting/changing a value, like when it's used for config vars.

Reply:
And I see there's no way to set the values in the context, so the user is expected to edit the files to change the context variables I assume?  Could `set` be used to set the vars?  eg: `fn context set api-url foo` ?

Reply:
I like the idea of both `set` and `use`, although we do have an `unset` command which links in with `set`. Does it make sense to have a `use` and `unset`?

An update command is coming where users can `fn context update <key> <value>` but for now they will edit it manually in their file. Also for the default context they can pass in the value `--api-url, --registry and --provider`.